### ISSUE TITLE
Refactor: add feature scaling 

### ISSUE THREAD

**Issue by sgreenbury** [alan-turing-institute/autoemulate#323]
Data processing is currently part of the pipeline and can affect performance substantially. There are various options for supporting with refactor:
- Have good default options such as:
  - standardizing inputs
  - no preprocessing (user handles any required preprocessing)
- Wrap with existing compatible processing packages (e.g. preprocessing modules in PyTorch)
- Implement a subset of commonly used transformations with compatible interface

**Comment by radka-j** [comment]
Once this refactor is finished, we should also close #247

**Comment by radka-j** [comment]
Just adding some notes here.

As mentioned, AutoEmulate wraps data processing (primarily standardisation but can include dimensionality reduction) and the model in an sklearn Pipeline object [here](https://github.com/alan-turing-institute/autoemulate/blob/main/autoemulate/model_processing.py). Specifically, the `_process_models` function is called on each model when [AutoEmulate.setup](https://github.com/alan-turing-institute/autoemulate/blob/752d942ed55cf66cda1659112df9c6cf0bbd38dc/autoemulate/compare.py#L118) is called.

Going forward, we might want to think about dimensionality reduction separately (see #348) and discuss here just the best way to handle feature scaling. The primary goal is to do away with the Pipeline wrapper and the associated sklearn scalers as they transform all data to numpy. 

We might have to simply implement our own Scaler objects, starting with the StandardScaler. Given input data, all it needs to do is store the training data mean and sd and then have 2 methods (transform and undo transform). Our model base classes then take this scalar as an optional input (this isn't all that different to what the interface looks like now). Down the line we can think about implementing other scalers if we find a use case for them.

**Comment by radka-j** [comment]
It's also worth noting that the GP and CNP modules already implement very simple normalisation by hand for the output values within the fit and predict methods and we could simply do the same thing for the input data as a starting point.

**Comment by sgreenbury** [comment]
Updated the title to reflect limiting the scope of this issue to scaling/normalization.

Next step: implement basic torch model that just scales the data with mean and std (standardizer)
